File: SuperAdmin/backend/SuperAdmin.API/Models/SubscriptionPlan.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SuperAdmin.API.Models;

[Table("subscription_plans")]
public class SubscriptionPlan
{
    [Key]
    [Column("id")]
    public Guid Id { get; set; }

    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [Column("price")]
    public decimal Price { get; set; }

    [Column("max_menu_items")]
    public int MaxMenuItems { get; set; }

    [Column("max_drivers")]
    public int MaxDrivers { get; set; }

    [Column("max_promotions")]
    public int MaxPromotions { get; set; }

    [Column("max_delivery_radius_km")]
    public int MaxDeliveryRadiusKm { get; set; }

    [Column("has_analytics")]
    public bool HasAnalytics { get; set; }

    [Column("has_custom_branding")]
    public bool HasCustomBranding { get; set; }

    [Column("has_inventory_export")]
    public bool HasInventoryExport { get; set; }

    [Column("commission_percent")]
    public decimal CommissionPercent { get; set; }

    [Column("features")]
    public string? Features { get; set; }

    // AI-gate columns (owned + seeded by the Spring backend via Flyway V55). Nullable because the
    // columns are nullable; copilot_monthly_quota = NULL means UNLIMITED (ENTERPRISE).
    [Column("has_promo_ai")]
    public bool? HasPromoAi { get; set; }

    [Column("has_driver_intel")]
    public bool? HasDriverIntel { get; set; }

    [Column("has_review_ai")]
    public bool? HasReviewAi { get; set; }

    [Column("has_api_access")]
    public bool? HasApiAccess { get; set; }

    [Column("copilot_monthly_quota")]
    public int? CopilotMonthlyQuota { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }
}

File: Backend/backend/src/main/java/com/example/backend/entity/SubscriptionPlan.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "subscription_plans")
public class SubscriptionPlan {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, unique = true)
    private String name;

    @Column(precision = 10, scale = 2)
    private BigDecimal price;

    @Column(nullable = false)
    private int maxMenuItems;

    @Column(nullable = false)
    private int maxDrivers;

    @Column(nullable = false)
    private int maxPromotions;

    @Column(nullable = false)
    private int maxDeliveryRadiusKm;

    @Column(nullable = false)
    private boolean hasAnalytics;

    @Column(nullable = false)
    private boolean hasCustomBranding;

    @Column(nullable = false)
    private boolean hasInventoryExport;

    // AI intelligence gates (nullable so ddl-auto can add them locally to the seeded plans;
    // null = not included). Promo AI / driver intelligence / review+support AI / API access.
    @Column(name = "has_promo_ai")
    private Boolean hasPromoAi;

    @Column(name = "has_driver_intel")
    private Boolean hasDriverIntel;

    @Column(name = "has_review_ai")
    private Boolean hasReviewAi;

    @Column(name = "has_api_access")
    private Boolean hasApiAccess;

    /** Monthly Copilot prompt quota (metering = margin protection); null = unlimited / fair use. */
    @Column(name = "copilot_monthly_quota")
    private Integer copilotMonthlyQuota;

    @Column(precision = 5, scale = 2)
    private BigDecimal commissionPercent;

    private String features;

    private LocalDateTime createdAt;
}
